### PR TITLE
Publish top information

### PR DESCRIPTION
**Pull request by DLu**
I want to be able to log the CPU usage over time, so I wrote a publisher. 

There needs to be a good way to turn the information on and off, but I didn't feel like mucking with the QT. 

CC: @dlaz

### CODE REVIEW

File: rqt_top/src/rqt_top/top_plugin.py [hunk 1]
```diff
@@ -34,6 +34,8 @@
 from python_qt_binding.QtCore import Qt, QTimer
 
 from rqt_top.node_info import NodeInfo
+from rqt_top.msg import *
```
Comment (dlaz) [inline rqt_top/src/rqt_top/top_plugin.py]: If I have one pet peeve about python style, it's `import *`. Please explicitly import what you need.

Comment (ablasdel) [inline rqt_top/src/rqt_top/top_plugin.py]: I agree with this sentiment. one should never import * it just makes things impossible to read.

**Comment by adolfo-rt** [comment]
> I want to be able to log the CPU usage over time, so I wrote a publisher. 

Only while `rqt` is running?, or would you like to log CPU usage in scenarios where there's potentially no X session?, or at least no running rqt instance?.

**Comment by DLu** [comment]
I'd like it without rqt, but I can use it either way.

**Comment by dlaz** [comment]
I agree it would be more generally useful without needing an X session.

File: rqt_top/msg/CPUUsage.msg [hunk 2]
```diff
@@ -0,0 +1 @@
+Process[] processes
```
Comment (dlaz) [inline rqt_top/msg/CPUUsage.msg]: Would these messages make more sense in diagnostic_msgs?

Comment (DLu) [inline rqt_top/msg/CPUUsage.msg]: @tfoote is the maintainer.

Comment (tfoote) [inline rqt_top/msg/CPUUsage.msg]: It's certainly something that could be considered standardizing. I'd suggest however to create a separate package with the cpu info, and maybe trying to add some similar info for the status of a computer such as memory info that can be a new package, and be reviewed and merged. The diagnostic messages are very specific to the diagnostic system, which is completely unrelated to cpu info.

**Comment by adolfo-rt** [comment]
@DLu, I don't know if you had in mind the extra overhead of exposing such resource info in a separate, non-rqt package; but I see great value in doing that.

**Comment by dlaz** [comment]
@adolfo-rt, I think it would be a good idea to pull out the actual node monitoring stuff into its own package where the functionality @DLu proposes can live. I'd be happy to make this happen, but it wouldn't be for at least a month or two.

**Comment by tfoote** [comment]
FYI if you're thinking about this the following paper might be of interest.

http://autonomylab.org/doc/monajjemi_crv14.pdf

**Comment by dirk-thomas** [comment]
Based on the feedback it sounds like it would make more sense to move this functionality into a non-rqt package in order to use it without any graphical component. Therefore I will close this. Please feel free to continue commenting.

**Pull request closed without merging**